internal/service/teams/resource_teams_test.go

```diff
@@ -14,6 +14,13 @@ import (
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
 )
 
+func TestMain(m *testing.M) {
+	acc.SetupSharedResources()
+	exitCode := m.Run()
+	acc.CleanupSharedResources()
+	os.Exit(exitCode)
+}
```

Comment:
regarding where TestMain should be placed within the test package: I do not see a clear convention other than ```place the TestMain function wherever it makes the most sense```.
With this in mind I believe having a `main_test.go` file (mentioned by @maastha) could be the most transparent option so it is clear the function impacts the entire test package. This is an example I found from the terraform repository: [e2etest/main_test.go](https://github.com/hashicorp/terraform/blob/ca2be46b0e4db9ab64e649da771d527dcd08c505/internal/command/e2etest/main_test.go#L29)

Reply:
@AgustinBettati @maastha ok, changed to their own main_test.go, also I like the idea of setup returning the cleanup function so cleanup doesn't need to be a public function

Reply:
Awesome!